modules/custom-post-types/portfolios.php

```diff
@@ -65,6 +65,14 @@ function __construct() {
 		add_shortcode( 'portfolio',                                                    array( $this, 'portfolio_shortcode' ) );
 		add_shortcode( 'jetpack_portfolio',                                            array( $this, 'portfolio_shortcode' ) );
 
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
```

Comment:
This way should almost certainly be faster-- checking for the existence of a constant should be about the most lightweight thing you can do